addons/survey/tests/test_survey.py

```diff
@@ -159,3 +159,150 @@ def test_skipped_values(self):
 
         for question in questions:
             self._assert_skipped_question(question, survey_user)
+
+    def test_get_pages_and_questions_to_show(self):
+        """
+        Tests the method `_get_pages_and_questions_to_show` - it takes a recordset of
+        question.question from a survey.survey and returns a recordset without
+        invalid conditional questions and pages without description
+
+        Structure of the test survey:
+
+        sequence    | type                          | trigger       | validity
+        ----------------------------------------------------------------------
+        1           | page, no description          | no trigger    | X
+        2           | text_box                      | trigger is 6  | X
+        3           | numerical_box                 | trigger is 2  | X
+        4           | simple_choice                 | no trigger    | V
+        5           | page, description             | no trigger    | V
+        6           | multiple_choice               | no trigger    | V
+        7           | multiple_choice, no answers   | no trigger    | V
+        8           | text_box                      | trigger is 6  | V
+        9           | matrix                        | trigger is 5  | X
+        10          | simple_choice                 | trigger is 7  | X
+        11          | simple_choice, no answers     | trigger is 8  | X
+        12          | text_box                      | trigger is 11 | X
+        """
+
+        my_survey = self.env['survey.survey'].create({
+            'title': 'my_survey',
+            'questions_layout': 'page_per_question',
+            'questions_selection': 'all',
+            'access_mode': 'public',
+        })
+        page_without_description = self.env['survey.question'].create({
```

Comment:
It's a bit confusing as you create questions in a different order that specified in the method docstring.
I think we could use a single create call to create all questions, without the conditional configuration.

It will look something like:

```py
[
    page_without_description,
    text_box_1,
    ...  
] = self.env['survey.question'].create([{
   ...
}, {
   ...
}])
```

And then manually call "write" on questions that need a conditional configuration afterwards.
It terms of performances it should be similar (1 big create + a few write VS a lot of small create) and it would be more readable.

For "survey.question.answer"s, as you don't need the reference, you could use ORM commands [(0, 0, values)].